File: 문제풀이/온라인 저지/프로그래머스/# 02. Level 2/# 132265. 롤케이크 자르기/kotlin/Solution.kt
class Solution {
    fun solution(topping: IntArray): Int {
        val counterA = HashMap<Int, Int>()
        val counterB = HashMap<Int, Int>()
        var answer = 0
        for (item in topping) {
            if (counterA.containsKey(item)) {
                counterA[item] = counterA[item]!! + 1
            } else {
                counterA[item] = 1
            }
        }
        for (item in topping) {
            counterA[item] = counterA[item]!! - 1
            if (counterA[item] == 0) counterA.remove(item)
            if (counterB.containsKey(item)) {
                counterB[item] = counterB[item]!! + 1
            } else {
                counterB[item] = 1
            }
            if (counterA.size == counterB.size) answer ++
        }
        return answer
    }
}

File: 문제풀이/온라인 저지/프로그래머스/# 02. Level 2/# 132265. 롤케이크 자르기/python/solution.py
from collections import Counter

def solution(topping):
    counter_a = Counter({})
    counter_b = Counter(topping)

    answer = 0

    for item in topping:
        if item not in counter_a:
            counter_a[item] = 1
        else:
            counter_a[item] += 1

        if counter_b[item] == 1:
            counter_b.pop(item)
        else:
            counter_b[item] -= 1

        if len(counter_a) == len(counter_b):
            answer += 1

    return answer
